src/v/cluster/health_monitor_backend.cc

```diff
@@ -151,16 +159,62 @@ std::optional<node_health_report> health_monitor_backend::build_node_report(
     return report;
 }
 
+void health_monitor_backend::abortable_refresh_request::abort() {
+    if (finished) {
+        return;
+    }
+    finished = true;
+    done.set_value(errc::timeout);
```

Comment:
As this isn't really a timeout, maybe we should invent a new error code like leader_changed -- then a higher level caller might choose to retry on that error.